src/tests/Loader/classloader/DefaultInterfaceMethods/regressions/github61244.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Before adding new tests, it might be worth checking whether this is already covered by the any of the existing default interface methods tests that are disabled on Mono. There's a ton of them and don't have an issue number:

https://github.com/dotnet/runtime/blob/6527b7fe83b95a195615c1081f345b6d34d716e7/src/tests/issues.targets#L1848-L1850

https://github.com/dotnet/runtime/blob/6527b7fe83b95a195615c1081f345b6d34d716e7/src/tests/issues.targets#L1913-L1918

https://github.com/dotnet/runtime/blob/6527b7fe83b95a195615c1081f345b6d34d716e7/src/tests/issues.targets#L1928-L1930

https://github.com/dotnet/runtime/blob/6527b7fe83b95a195615c1081f345b6d34d716e7/src/tests/issues.targets#L1934-L1939

https://github.com/dotnet/runtime/blob/6527b7fe83b95a195615c1081f345b6d34d716e7/src/tests/issues.targets#L1943-L1945

Reply:
(This test look pretty low value because it doesn't try to do a dispatch or check whether we have a good generic context.)

Reply:
I did check and none of the existing tests you list cover this case. If you want a more interesting case I can push this following...  

```
// The .NET Foundation licenses this file to you under the MIT license.

using System;

class Program
{
    static int Main(string[] args)
    {
        return new TestClass().DoTest();
    }
}

public interface OneArgBaseInterface<T1>
{
    int SomeFunc1(T1 someParam1, Type someParam1Type);
}

public interface TwoArgBaseInterface<T1, T2>
{
    int SomeFunc1(T1 someParam1, T2 someParam2, Type someParam1Type, Type someParam2Type);
}

public interface SecondInterface<TParam2Type, TParam1Type> :  OneArgBaseInterface<TParam1Type>, TwoArgBaseInterface<TParam1Type, TParam2Type>
{
    int OneArgBaseInterface<TParam1Type>.SomeFunc1(TParam1Type someParam1, Type someParam1Type)
    {
        if (typeof(TParam1Type) != someParam1Type)
        {
            Console.WriteLine("Failed => 101");
            return 101;
        }

        return 100;
    }

    int TwoArgBaseInterface<TParam1Type, TParam2Type>.SomeFunc1(TParam1Type someParam1, TParam2Type someParam2, Type someParam1Type, Type someParam2Type)
    {
        if (typeof(TParam1Type) != someParam1Type)
        {
            Console.WriteLine("Failed => 102");
            return 102;
        }

        if (typeof(TParam2Type) != someParam2Type)
        {
            Console.WriteLine("Failed => 103");
            return 103;
        }

        return 100;
    }
}

public class TestClass : SecondInterface<int, string>
{
    public int DoTest ()
    {
        int ret = (this as OneArgBaseInterface<string>).SomeFunc1("test string", typeof(string));
        if (ret != 100)
            return ret;

        ret = (this as TwoArgBaseInterface<string, int>).SomeFunc1("test string", 0, typeof(string), typeof(int));
        if (ret != 100)
            return ret;

        Console.WriteLine("Passed => 100");
        return 100;
    }
}
```

Reply:
👍 this test looks much more thorough! Thanks!